tsl/test/sql/cagg_query.sql

```diff
@@ -296,3 +296,641 @@ ORDER BY m1.location COLLATE "C" NULLS LAST, m1.timec DESC NULLS LAST, firsth NU
+                         '2000-01-01 23:59:59 PST','1m') time;
+
+INSERT INTO temperature
+  SELECT time, 6
+    FROM generate_series('2020-01-01 00:00:00 PST'::timestamptz,
+                         '2020-01-01 23:59:59 PST','1m') time;
+
+-- Create CAggs
+CREATE MATERIALIZED VIEW cagg_4_hours
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+
+CREATE MATERIALIZED VIEW cagg_4_hours_offset
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time, '30m'::interval), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+
+-- Align origin with first value
+CREATE MATERIALIZED VIEW cagg_4_hours_origin
+  WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+  SELECT time_bucket('4 hour', time, '2000-01-01 01:00:00 PST'::timestamptz), max(value)
+    FROM temperature
+    GROUP BY 1 ORDER BY 1;
+
+-- Query the CAggs and check that all buckets are materialized
+SELECT time_bucket('4 hour', time), max(value) FROM temperature GROUP BY 1 ORDER BY 1;
+SELECT * FROM cagg_4_hours;
+ALTER MATERIALIZED VIEW cagg_4_hours SET (timescaledb.materialized_only=true);
+SELECT * FROM cagg_4_hours;
+
+SELECT time_bucket('4 hour', time, '30m'::interval), max(value) FROM temperature GROUP BY 1 ORDER BY 1;
+SELECT * FROM cagg_4_hours_offset;
+ALTER MATERIALIZED VIEW cagg_4_hours_offset SET (timescaledb.materialized_only=true);
+SELECT * FROM cagg_4_hours_offset;
+
+SELECT time_bucket('4 hour', time, '2000-01-01 01:00:00 PST'::timestamptz), max(value) FROM temperature GROUP BY 1 ORDER BY 1;
+SELECT * FROM cagg_4_hours_origin;
+ALTER MATERIALIZED VIEW cagg_4_hours_origin SET (timescaledb.materialized_only=true);
+SELECT * FROM cagg_4_hours_origin;
+
+-- Update the last bucket and re-materialize
+INSERT INTO temperature values('2020-01-01 23:55:00 PST', 10);
+
+CALL refresh_continuous_aggregate('cagg_4_hours', NULL, NULL);
+CALL refresh_continuous_aggregate('cagg_4_hours_offset', NULL, NULL);
+CALL refresh_continuous_aggregate('cagg_4_hours_origin', NULL, NULL);
+
+SELECT * FROM cagg_4_hours;
+SELECT * FROM cagg_4_hours_offset;
+SELECT * FROM cagg_4_hours_origin;
+
+-- Check the real-time functionality
+ALTER MATERIALIZED VIEW cagg_4_hours SET (timescaledb.materialized_only=false);
+ALTER MATERIALIZED VIEW cagg_4_hours_offset SET (timescaledb.materialized_only=false);
+ALTER MATERIALIZED VIEW cagg_4_hours_origin SET (timescaledb.materialized_only=false);
+
+-- Check watermarks
+SELECT *, _timescaledb_functions.to_timestamp(watermark)
+  FROM _timescaledb_catalog.continuous_aggs_watermark
+  WHERE mat_hypertable_id IN (SELECT id FROM _timescaledb_catalog.hypertable
+    WHERE table_name IN (
+      SELECT materialization_hypertable_name
+        FROM timescaledb_information.continuous_aggregates
+        WHERE view_name LIKE 'cagg_4_hours%'
+    )
+  )
+;
```

Comment:
```suggestion
SELECT continuous_aggs_watermark.*, _timescaledb_functions.to_timestamp(watermark)
  FROM _timescaledb_catalog.continuous_aggs_watermark
  JOIN _timescaledb_catalog.continuous_agg USING (mat_hypertable_id)
WHERE user_view_name LIKE 'cagg_4_hours%';
```
Just tryin to make it simpler... WDYT?

Reply:
This query looks much cleaner, changed it. Thank you.